mne/viz/epochs.py

```diff
@@ -88,27 +90,109 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
 
-    picks = np.atleast_1d(picks)
-    if (fig is not None or axes is not None) and len(picks) > 1:
-        raise ValueError('Only single pick can be drawn to a figure.')
+    ch_types = [channel_type(epochs.info, idx) for idx in picks]
+    for ch_type in ch_types:
+        if ch_type not in scalings:
+            # We know it's not in either scalings or units since keys match
+            raise KeyError('%s type not in scalings and units' % ch_type)
+
+    # First, we collect groupings of picks and types in two lists
+    # (all_picks, all_ch_types) -> groupby.
+    # Then, we construct a list of the corresponding data, names and evokeds
+    # (to_plot_list) -> combine.
+    # Then, we loop over this list and plot using _plot_epochs_image.
+
+    # groupby
+    if groupby is None:
+        all_picks, all_ch_types = [picks], ch_types   # fix!!!!
+    else:
+        if groupby == "type":
+            all_picks, all_ch_types = list(), list()
+            for this_type in set(ch_types):
+                these_picks = picks[np.array(ch_types) == this_type]
+                all_picks.append(these_picks)
+                all_ch_types.append(this_type)
+        if isinstance(groupby, dict):
+            if combine is None:
+                msg = "If `groupby` is a dict, `combine` must not be None"
+                raise ValueError(msg)
+            all_ch_types = list(groupby.keys())
+            all_picks = [groupby[ch_type] for ch_type in all_ch_types]
+
+    # combine/construct list for plotting
+    data = epochs.get_data()
+    to_plot_list = list()
+    if combine is None:
+        for pick, ch_type in zip(picks, all_ch_types):
+            this_data = data[:, pick, :].squeeze() #* scalings[ch_type] # (n_epochs, n_channels, n_times)
+            to_plot_list.append((this_data, ch_type, epochs.average(picks=[pick]),
+                                 epochs.ch_names[pick]))
+    else:
+        name_dict = {"eeg": "EEG", "grad": "Gradiometers",
+                     "mag": "Magnetometers"}
+        if combine == "gfp":
+            combine = lambda D: np.sqrt((D * D).mean(axis=1))
+            vmin = 0
+        elif combine == "mean":
+            combine = lambda D: np.mean(D, 1)
+        elif combine == "std":
+            combine = lambda D: np.std(D, 1)
+            vmin = 0
+        elif not callable(combine):
+            tmplt = ("`combine` must be None, a callable or "
+                     "one out of 'mean' or 'gfp`. Got {}.")
+            raise ValueError(tmplt.format(type(combine)))
+        for ch_type, picks_ in zip(all_ch_types, all_picks):
+            this_data = combine(data[:, picks_, :]) / scalings[ch_type] #* scalings[ch_type] # (n_epochs, n_channels, n_times)
+            ev = epochs.average(picks=picks_)
+            ev.data = this_data.mean(0)
+            to_plot_list.append((this_data, ch_type, ev,
+                                 name_dict.get(ch_type, ch_type)))
```

Comment:
please create a subfunction that is called _combine_data and that returns to_plot_list